### PR TITLE
[Subjects] Add search input and title

### LINKED ISSUE #196: (SP: 1) [Subjects] Add search input and title
UserStory: #66 
## Description
Add title with description, implement input with searching functionality to search for a subject by name and/or category
## Design 
image
## Acceptance Criteria
- [ ] When a user clicks on the "Show all offers", he/she should be redirected to "find offers" page
- [ ] When a user clicks on the "Back to all categories", he/she should be redirected to "categories" page
- [ ] When a user selects category from category menu, all subjects that are related to that category should be displayed
- [ ] When a user clicks on the "Search" button or press enter, search should be performed

### PR DESCRIPTION
**Pull request by YaroslavChuiko**
- [x] When a user clicks on the "Show all offers", he/she should be redirected to "find offers" page
- [x]  When a user clicks on the "Back to all categories", he/she should be redirected to "categories" page
- [x]  When a user selects category from category menu, all subjects that are related to that category should be displayed
- [x]  When a user clicks on the "Search" button or press enter, search should be performed

### Desktop
image
### Mobile
image
### Search params when the form is filled
image
### Test coverage
image

### CODE REVIEW

**Review by holus-bolus — APPROVED** [review]

**Review by OlhaOvsiienko — APPROVED** [review]

**Review by YaroslavLys — APPROVED** [review]

**Review by khytrykv — APPROVED** [review]

File: src/containers/subjects-search/SubjectsSearch.jsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import { authRoutes } from '~/router/constants/authRoutes'
+import { categoryService } from '~/services/category-service'
+
+const SubjectsSearch = () => {
+  const { t } = useTranslation()
+  const { isMobile } = useBreakpoints()
+  const [searchParams, setSearchParams] = useSearchParams()
+  const [subjectName, setSubjectName] = useState(
+    searchParams.get('subjectName') || ''
+  )
```
Comment (vlad-khrychov) [inline src/containers/subjects-search/SubjectsSearch.jsx]: https://react.dev/reference/react/useState#avoiding-recreating-the-initial-state
```suggestion
  const [subjectName, setSubjectName] = useState(
    ()=>(searchParams.get('subjectName') || '')
  )
```

File: src/containers/subjects-search/SubjectsSearch.jsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      setSearchParams((params) => {
+        params.delete('categoryId')
+        params.delete('categoryName')
+        return params
+      })
+      return
+    }
+    setSearchParams((params) => {
+      params.set('categoryId', categoryValue._id)
+      params.set('categoryName', categoryValue.name)
```
Comment (vlad-khrychov) [inline src/containers/subjects-search/SubjectsSearch.jsx:37]: Why do we need both ID and name in the URL?

File: src/containers/subjects-search/SubjectsSearch.jsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+          />
+        </Box>
+        {!isMobile && (
+          <AppButton type='submit' variant='containedLight'>
+            {t('common.search')}
+          </AppButton>
+        )}
+      </Box>
+      {isMobile && categoryInput}
+      {!isMobile && footer}
```
Comment (vlad-khrychov) [inline src/containers/subjects-search/SubjectsSearch.jsx]:
```suggestion
      {isMobile ? categoryInput : footer}
```

File: tests/unit/containers/subjects-search/SubjectsSearch.spec.jsx
Comment (vlad-khrychov) [inline tests/unit/containers/subjects-search/SubjectsSearch.spec.jsx:1]: @Space2Study-UA-1156/team-01 here is good example of tests with mock of react router.

**Review by vlad-khrychov — APPROVED** [review]

File: src/containers/subjects-search/SubjectsSearch.jsx (hunk 2 above)
Comment (YaroslavChuiko) [reply src/containers/subjects-search/SubjectsSearch.jsx:37]: I use categoryName to set the block title. If it is better to save it in the component state, I can try to fix that

image

**Pull request merged**